src/wasm-binary-writer-spec.c

```diff
@@ -284,7 +284,8 @@ static WasmFunc* append_nullary_func(WasmAllocator* allocator,
   WasmModuleField* func_field =
       append_module_field(allocator, module, WASM_MODULE_FIELD_TYPE_FUNC);
   WasmFunc* func = &func_field->func;
-  func->decl.flags = WASM_FUNC_DECLARATION_FLAG_HAS_FUNC_TYPE;
+  func->decl.flags = WASM_FUNC_DECLARATION_FLAG_HAS_FUNC_TYPE |
+                     WASM_FUNC_DECLARATION_FLAG_SHARED_SIGNATURE;
```

Comment:
the shared flag is set whenever the signature is owned by the module. This is true most of the time; the only time it's not is when the signature is first being created inside the parser, and has been added to the module yet. It's a bit of a hack, honestly, and I've refactored it a few times but haven't really made it much cleaner. I think I know how to clean it up, but it would be a bigger change, and probably shouldn't be tackled until the tests all pass.